diff --git a/flash_head/src/modules/flash_head_model.py b/flash_head/src/modules/flash_head_model.py
index 989cd7c..a4825a0 100755
--- a/flash_head/src/modules/flash_head_model.py
+++ b/flash_head/src/modules/flash_head_model.py
@@ -1,3 +1,4 @@
+import os
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
@@ -34,7 +35,7 @@ except ModuleNotFoundError:
     
     
 def flash_attention(q: torch.Tensor, k: torch.Tensor, v: torch.Tensor, num_heads: int, compatibility_mode=False):
-    if compatibility_mode:
+    if compatibility_mode or os.getenv("FLASHHEAD_T4") == "1":
         q = rearrange(q, "b s (n d) -> b n s d", n=num_heads)
         k = rearrange(k, "b s (n d) -> b n s d", n=num_heads)
         v = rearrange(v, "b s (n d) -> b n s d", n=num_heads)
diff --git a/flash_head/src/pipeline/flash_head_pipeline.py b/flash_head/src/pipeline/flash_head_pipeline.py
index 6d0b757..684a383 100644
--- a/flash_head/src/pipeline/flash_head_pipeline.py
+++ b/flash_head/src/pipeline/flash_head_pipeline.py
@@ -16,8 +16,8 @@ from flash_head.utils.utils import match_and_blend_colors_torch, resize_and_cent
 from flash_head.utils.facecrop import process_image
 
 # compile models to speedup inference
-COMPILE_MODEL = True
-COMPILE_VAE = True
+COMPILE_MODEL = os.getenv("FLASHHEAD_T4") != "1"
+COMPILE_VAE = os.getenv("FLASHHEAD_T4") != "1"
 # use parallel vae to speedup decode/encode, only support WanVAE
 USE_PARALLEL_VAE = True
 
@@ -74,6 +74,8 @@ class FlashHeadPipeline:
             use_usp (`bool`, *optional*, defaults to False):
                 Enable distribution strategy of USP.
         """
+        if os.getenv("FLASHHEAD_T4") == "1":
+            param_dtype = torch.float16
         self.param_dtype = param_dtype
         self.device = device
         self.rank = dist.get_rank() if dist.is_initialized() else 0
